File: shablon.cs
using System;

interface IDatabaseConnection
{
    void Open();
    void Close();
}

interface IDatabaseCommand
{
    void Execute(string query);
}

interface IDatabaseTransaction
{
    void Begin();
    void Commit();
    void Rollback();
}


class MySQLConnection : IDatabaseConnection
{
    public void Open()
    {
        Console.WriteLine("MySQL базасына қосылды");
    }

    public void Close()
    {
        Console.WriteLine("MySQL базасы жабылды");
    }
}


class MySQLCommand : IDatabaseCommand
{
    public void Execute(string query)
    {
        Console.WriteLine("Сұраныс орындалды: " + query);
    }
}


class MySQLTransaction : IDatabaseTransaction
{
    public void Begin()
    {
        Console.WriteLine("Транзакция басталды");
    }

    public void Commit()
    {
        Console.WriteLine("Транзакция сақталды");
    }

    public void Rollback()
    {
        Console.WriteLine("Транзакция қайтарылды");
    }
}


class MySQLFactory
{
    public IDatabaseConnection CreateConnection()
    {
        return new MySQLConnection();
    }

    public IDatabaseCommand CreateCommand()
    {
        return new MySQLCommand();
    }

    public IDatabaseTransaction CreateTransaction()
    {
        return new MySQLTransaction();
    }
}


public class Program
{
    public static void Main(string[] args)
    {
        MySQLFactory factory = new MySQLFactory();

        IDatabaseConnection connection = factory.CreateConnection();
        IDatabaseCommand command = factory.CreateCommand();
        IDatabaseTransaction transaction = factory.CreateTransaction();

        connection.Open();

        transaction.Begin();

        command.Execute("SELECT * FROM users");

        transaction.Commit();

        connection.Close();
    }
}

File: shablon.py

class IDatabaseConnection:
    def open(self):
        pass

    def close(self):
        pass


class IDatabaseCommand:
    def execute(self, query):
        pass


class IDatabaseTransaction:
    def begin(self):
        pass

    def commit(self):
        pass

    def rollback(self):
        pass


class MySQLConnection(IDatabaseConnection):
    def open(self):
        print("MySQL базасына қосылды")

    def close(self):
        print("MySQL базасы жабылды")


class MySQLCommand(IDatabaseCommand):
    def execute(self, query):
        print("Сұраныс орындалды:", query)


class MySQLTransaction(IDatabaseTransaction):
    def begin(self):
        print("Транзакция басталды")

    def commit(self):
        print("Транзакция сақталды")

    def rollback(self):
        print("Транзакция қайтарылды")


class MySQLFactory:
    def create_connection(self):
        return MySQLConnection()

    def create_command(self):
        return MySQLCommand()

    def create_transaction(self):
        return MySQLTransaction()


factory = MySQLFactory()

connection = factory.create_connection()
command = factory.create_command()
transaction = factory.create_transaction()

connection.open()

transaction.begin()

command.execute("SELECT * FROM users")

transaction.commit()

connection.close()
